ivy/functional/backends/paddle/experimental/set.py

```diff
@@ -0,0 +1,57 @@
+# global
+from typing import Tuple
+import paddle
+from ivy.func_wrapper import with_unsupported_device_and_dtypes
+
+# local
+import ivy.functional.backends.paddle as paddle_backend
+from . import backend_version
+
+
+@with_unsupported_device_and_dtypes(
+    {"2.4.2 and below": {"cpu": ("complex", "int8", "uint8", "int16", "float16")}},
+    backend_version,
+)
+def intersection(
+    x1: paddle.Tensor,
+    x2: paddle.Tensor,
+    /,
+    *,
+    assume_unique: bool = False,
+    return_indices: bool = False,
+) -> Tuple[paddle.Tensor, paddle.Tensor, paddle.Tensor]:
+    x1 = paddle.reshape(x1, [-1])
+    x2 = paddle.reshape(x2, [-1])
+    if not assume_unique:
+        x1, ind1, _, _ = paddle_backend.unique_all(x1)
+        x2, ind2, _, _ = paddle_backend.unique_all(x2)
+
+    aux = paddle.concat([x1, x2], 0)
+    if return_indices:
+        values_ = paddle.moveaxis(aux, 0, 0)
```

Comment:
what does this line do?

Reply:
Line 31 to 35 is basically a replacement for `paddle.argsort` which does not preserve the order of equivalent elements. For instance `torch.argsort` provides the `stable` param to enforce this. Without this, I observed that the results are inconsistent.

Reply:
I see, but line 31 seems to do nothing, you are swapping 0th axis with 0th axis so the resultant array is equal to the input